# Experiment dd939c4e9dab54d886c1857b9b104298fe0b66dba4c60d14e0e9ecdb152337f2
# Base 4a603c669103afff6e2ff8ff6ab2edc399b3cb42
diff --git a/src/Mouseptr.cpp b/src/Mouseptr.cpp
index 029e478a6..b0b30d25f 100644
--- a/src/Mouseptr.cpp
+++ b/src/Mouseptr.cpp
@@ -586,12 +586,12 @@ void TMousePointer::set_facet(int f)
     this->d_bottom = this->cHeight - info->Hotspot_Y;
 }
 
-void TMousePointer::set_game_facet(int facet)
+void TMousePointer::set_game_facet(int facet_in)
 {
-    if (facet >= 0 && facet < this->Num_Cursors) {
-        Shape_Info* info = this->Cursor_Shape->shape_info;
-        this->game_facet = facet;
-        info = info + facet;
+    if (facet_in >= 0 && facet_in < this->Num_Cursors) {
+        TShape* shape = this->Cursor_Shape;
+        this->game_facet = facet_in;
+        Shape_Info* info = shape->shape_info + facet_in;
         this->gWidth = info->Width;
         this->gHeight = info->Height;
         this->g_top = info->Hotspot_Y;
